Global Sensitivity Analysis

abachma2 opened the pull request:
This PR adds work done to perform global sensitivity analysis for the transition from US LWRs to advanced reactors. The ``README.md`` file explains some of the file structure, so I recommend reviewing that first. I figured a ``README`` was a better place to store that information that just a PR. 

Each subdirectory in the ``global`` directory contains four files: ``list_sample.in``, ``dakota.in`` (both are Dakota input files, see README for difference), ``*_driver.py`` (driver file to create Cyclus input files), and ``*_input.xml.in`` (Template for Cyclus input files).

nsryan2 commented on input/haleu/sensitivity-analysis/global/README.md:
```diff
@@ -0,0 +1,24 @@
+Xe-100 burnup, and MMR build share are varied, but the build share 
+of only advanced reactor is varied. The build share of only 
+one advanced reactor is varied at a time to prevent unphysical 
+combinations of advanced reactor build shares (e.g. 50% Xe-100, 
```
```suggestion
combinations of advanced reactor build shares (e.g., 50% Xe-100, 
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/dakota.in:
```diff
@@ -0,0 +1,43 @@
+    active all
+    uniform_uncertain = 5
+    lower_bounds = 721  0     0            41           28
+    upper_bounds  901   50    50           90           185
```
```suggestion
    upper_bounds =  901   50    50           90           185
```
Does this variable need an equal sign?

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/dakota.in:
```diff
@@ -0,0 +1,43 @@
+    uniform_uncertain = 5
+    lower_bounds = 721  0     0            41           28
+    upper_bounds  901   50    50           90           185
+    descriptors   'ts' 'lwr' 'mmr_share' 'mmr_burnup' 'xe100_burnup'   
```
```suggestion
    descriptors =    'ts' 'lwr' 'mmr_share' 'mmr_burnup' 'xe100_burnup'   
```
This one too? I'm not super familiar with this type of input file, just trying to match the pattern

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/list_sample.in:
```diff
@@ -0,0 +1,40 @@
+variables,
+    active all 
+    continuous_design = 4 
+      lower_bounds  721   0     0            41
```
```suggestion
      lower_bounds =  721   0     0            41
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/list_sample.in:
```diff
@@ -0,0 +1,40 @@
+    active all 
+    continuous_design = 4 
+      lower_bounds  721   0     0            41
+      upper_bounds  901   50    50           90 
```
```suggestion
      upper_bounds =  901   50    50           90 
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/mmr_share_driver.py:
```diff
@@ -0,0 +1,97 @@
+
+# Edit Cyclus input file
+cyclus_template = 'mmr_share_input.xml.in'
+scenario_name = ('ts_' + str(int(params['ts'])) +
```
I could be understanding the variables incorrectly, but are you worried about losing some information by turning 'mmr_share,' 'xe100_burnup,' or 'mmr_burnup' into ints (in my mind they could be floats).

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/mmr_share_input.xml.in:
```diff
@@ -0,0 +1,248 @@
+          <val>cooled_spent_mmr_haleu</val>
+        </out_commods>
+        <residence_time>1</residence_time>
+        <latitude>36.8286</latitude>
```
```suggestion
        <!-- Yucca Mountain Repository -->
        <latitude>36.8286</latitude>
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/mmr_share_input.xml.in:
```diff
@@ -0,0 +1,248 @@
+          <val>cooled_spent_smr_fuel</val>
+        </out_commods>
+        <residence_time>1</residence_time>
+        <latitude>36.8286</latitude>
```
```suggestion
        <!-- Yucca Mountain Repository -->
        <latitude>36.8286</latitude>
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/mmr_share/mmr_share_input.xml.in:
```diff
@@ -0,0 +1,248 @@
+    </config>
+  </facility>
+  <facility>
+    <name>Xe-100CoolingPool</name>
```
(A half baked question about the location of the cooling pools): Is this saying that you're deploying an MMR, VOYGR, and xe100 at Yucca mountain? Or is that not what these pools are?

nsryan2 commented on input/haleu/sensitivity-analysis/global/voygr_share/dakota.in:
```diff
@@ -0,0 +1,43 @@
+    active all
+    uniform_uncertain = 5
+    lower_bounds = 721  0     0            41           28
+    upper_bounds  901   50    50           90           185
```
```suggestion
    upper_bounds = 901   50    50           90           185
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/voygr_share/dakota.in:
```diff
@@ -0,0 +1,43 @@
+    uniform_uncertain = 5
+    lower_bounds = 721  0     0            41           28
+    upper_bounds  901   50    50           90           185
+    descriptors   'ts' 'lwr' 'voygr_share' 'mmr_burnup' 'xe100_burnup'   
```
```suggestion
    descriptors  = 'ts' 'lwr' 'voygr_share' 'mmr_burnup' 'xe100_burnup'   
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/voygr_share/list_sample.in:
```diff
@@ -0,0 +1,40 @@
+variables,
+    active all 
+    continuous_design = 4 
+      lower_bounds  721   0     0            41
```
```suggestion
      lower_bounds = 721   0     0            41
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/xe100_share/dakota.in:
```diff
@@ -0,0 +1,43 @@
+    uniform_uncertain = 5
+    lower_bounds = 721  0     0            41           28
+    upper_bounds  901   50    50           90           185
+    descriptors   'ts' 'lwr' 'xe100_share' 'mmr_burnup' 'xe100_burnup'   
```
```suggestion
    descriptors  = 'ts' 'lwr' 'xe100_share' 'mmr_burnup' 'xe100_burnup'   
```

nsryan2 commented on input/haleu/sensitivity-analysis/global/xe100_share/list_sample.in:
```diff
@@ -0,0 +1,40 @@
+variables,
+    active all 
+    continuous_design = 4 
+      lower_bounds  721   0     0            41
```
```suggestion
      lower_bounds = 721   0     0            41
```

nsryan2 reviewed (approved):
Overall, looks good to me! My only comment about the jupyter notebook was that the plot titles are all the same, but that's easily changeable and that could easily be a later addition or something you were still working on. In the input files, I noticed some of the equal signs are inconsistent and Idk if that's allowed (I mean, it worked so I guess it's allowed).

abachma2 commented:
Thanks for the review @nsryan2. I have addressed some of the suggestions you made. I added more equal signs in the Dakota input files to be more consistent and improve readability. I'm not sure what you mean by the titles of the graphs in the notebook. They vary by section and reflect the advanced reactor build share varied. The x and y labels then describe the parameter and metric. What do you think the titles should be. I am finished with the runs for this work, so any changes to this code should be included in this PR.

nsryan2 commented:
Ok, great! I wasn't sure if the equals signs were important, but they could help readability. 

I didn't run the notebook (just looked at the output that was already there) and I think the titles could be improved by saying which parameters and metrics (even though they're on the axes) are varying. I'm just thinking if I were reading a draft of a paper with these in it, the titles would stand out as being too general. And, this might only apply to "ENR_U," some of the axis titles could be clearer if you spelled it out in the title (also what the caption could be for, but obviously you're not going to have that in a notebook).

abachma2 commented:
I see what you mean. I have updated the figure and axis titles. Let me know if you think anything else needs to be done.

abachma2 commented:
@nsryan2 bump

nsryan2 commented:
Looking much better on the plots! Idk if this is doable, but the scale (the little 1e#) thing is interfering with some of the titles, can you move that?

abachma2 commented:
There is a way to do that. I have adjusted the plot titles up some so it is all more readable.

nsryan2 commented:
Great! All LGTM! Merging
